src/Moonglade.Web/Controllers/PostController.cs

```diff
@@ -90,6 +90,33 @@ public async Task<IActionResult> Slug(int year, int month, int day, string slug)
             return View(viewModel);
         }
 
+        [Route("{year:int:min(1975):length(4)}/{month:int:range(1,12)}/{day:int:range(1,31)}/{slug:regex(^(?!-)([[a-zA-Z0-9-]]+)$)}/{raw:regex(^(meta|content)$)}")]
+        public async Task<IActionResult> Raw(int year, int month, int day, string slug, string raw)
+        {
+            if (year > DateTime.UtcNow.Year || string.IsNullOrWhiteSpace(slug))
+            {
+                Logger.LogWarning($"Invalid parameter year: {year}, slug: {slug}");
+                return NotFound();
+            }
+
+            switch (raw.ToLower())
+            {
+                case "meta":
+                    var rspMeta = await _postService.GetPostMetaAsync(year, month, day, slug);
+                    return !rspMeta.IsSuccess 
+                        ? ServerError(rspMeta.Message) 
+                        : Json(rspMeta.Item);
+
+                case "content":
+                    var rspContent = await _postService.GetPostRawContentAsync(year, month, day, slug);
+                    return !rspContent.IsSuccess 
+                        ? ServerError(rspContent.Message) 
+                        : Content(rspContent.Item, "text/plain");
```

Comment:
Couldn't the MIME type be `text/markdown; charset=UTF-8`?

https://stackoverflow.com/questions/10701983/what-is-the-mime-type-for-markdown

Reply:
Because the content isn't always markdown. ```text/plain``` is for compatibility